app/src/main/java/com/example/tasky/feature_agenda/presentation/datepicker/MonthPickerOnToolbar.kt

```diff
@@ -0,0 +1,229 @@
+import androidx.compose.material3.Icon
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextButton
+import androidx.compose.material3.rememberDatePickerState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.sp
+import com.example.tasky.R
+import com.example.tasky.common.domain.util.convertMillisToMonth
+import com.vanpra.composematerialdialogs.MaterialDialog
+import com.vanpra.composematerialdialogs.MaterialDialogState
+import com.vanpra.composematerialdialogs.datetime.date.datepicker
+import com.vanpra.composematerialdialogs.rememberMaterialDialogState
+
+@Composable
+fun MonthTextWithIcon(
+    monthSelected: String,
+    onClick: () -> Unit
+) {
+    Row(
+        modifier = Modifier
+            .background(color = Color.Black)
+            .clickable { onClick() }
+    ) {
+        Text(
+            text = monthSelected,
+            color = Color.White,
+            fontWeight = FontWeight.Bold,
+            fontSize = 19.sp
+        )
+        Icon(
+            imageVector = Icons.Default.ArrowDropDown,
+            contentDescription = null,
+            modifier = Modifier
+                .background(Color.Black),
+            tint = Color.White
+        )
+    }
+}
+
+@Composable
+@Preview
+fun PreviewMonthIconSelectionText() {
+    MonthTextWithIcon(
+        monthSelected = "MARCH",
+        onClick = { }
+    )
+}
+
+@Composable
+fun MonthPickerOnToolbar(
+    monthSelected: String,
+    onMonthSelected: (String) -> Unit,
+    dialogState: MaterialDialogState,
+    onDialogStateChange: (MaterialDialogState) -> Unit
+) {
+
+    // Button to open the DatePickerDialog
+    MonthTextWithIcon(
+        monthSelected = monthSelected,
+        onClick = {
+            dialogState.show()
+            onDialogStateChange(dialogState)
+        }
+    )
+
+    MaterialDialog(
+        dialogState = dialogState,
+        buttons = {
+            positiveButton(
+                stringResource(id = R.string.ok),
+                textStyle = TextStyle(color = Color.Black)
+            )
+            negativeButton(
+                stringResource(id = R.string.cancel),
+                textStyle = TextStyle(color = Color.Black)
+            )
+        }
+    ) {
+        datepicker(
+            yearRange = IntRange(2023, 2030),
+            colors = customDatePickerColors()
+        ) { date ->
+            onMonthSelected(date.month.toString())
+            dialogState.hide()
+            onDialogStateChange(dialogState)
+        }
```

Comment:
Here you can notice that this date picker library also works with the DateTime library. The `date` is of type `LocalDate`. I talk about the most important parts of it here:
https://youtu.be/gzHy6wKAJh8?si=sAulwX7i8Go52Ws4